nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -144,38 +124,45 @@ def run(self, data=None, commit=None):
 
         # Loop over each of the circuit maintenance records
         # pylint: disable=too-many-nested-blocks
-        counter: int = 0
+        processed_maintenance_list = []  # List of the processed maintenance records
         for circuit_maint in circuit_maintenances:
-            counter += 1
-            # Get the list of sites
-            site_list: list = []
+            # Get the set of sites
+            site_set: set = set()
             for circuit in circuit_maint.circuits:
-                site_list += get_sites_from_circuit(circuit)
+                site_set.update(get_sites_from_circuit(circuit))
 
+            processed_maintenance_list.append(circuit_maint)
+            overlapping_maintenance = False  # Flag for overlapping maintenance present
             # Check to see if there are any circuit maintenances that are overlapping at some moment in time
-            for site in site_list:
+            for site in site_set:
                 for other_circuit_maint in circuit_maintenance_mapper[site.name]:
                     # Report failures for any time where a circuit will take an outage
-                    if circuit_maint == other_circuit_maint:
+                    # Check to see if the other circuit maintenance has already had overlap checked.
+                    # If this is the same circuit maintenance record, that will also show up in the processed list
+                    # since the list is being added to before the loop.
+                    if other_circuit_maint in processed_maintenance_list:
                         continue
                     if check_for_overlap(circuit_maint, other_circuit_maint):
+                        overlapping_maintenance = True
                         if PLUGIN_SETTINGS.get("overlap_job_exclude_no_impact"):
                             for impact in circuit_maint.circuitimpact_set:
                                 if impact.impact != "NO-IMPACT":
                                     self.log_warning(
                                         obj=site,
-                                        message=f"There is an overlapping maintenance for site: {site.name} on {circuit_maint.start_time}. Other maintenances: {other_circuit_maint}|{circuit_maint}",
+                                        message=f"There is an overlapping __no-impact__ maintenance for site: {site.name} on {circuit_maint.start_time}. Other maintenances: {other_circuit_maint}|{circuit_maint}",
```

Comment:
I believe that this is wrong. if we are here, this impact is NOT (!=) "NO-IMPACT", so it's a regular overlapping. Maybe it would be good to add a DEBUG log, for the "NO-IMPACT", saying that there is overlap, but becuase it has no impact, we simply log it, without complaining.

Reply:
This gets back into the discussion about `NO-IMPACT` notifications and the setting that is in place to address this. In my experience I would always want to warn about any notifications when it came to circuits, even if one is sent out for `NO-IMPACT`. It's like a change request that has no **expected** impact. But there is risk because something is being done.